### PR TITLE
Webaudio

### PR DESCRIPTION
**Pull request by sigvef**

### CODE REVIEW

**Comment by sigvef** [comment]
ping

File: nin/dasBoot/Loader.js [hunk 1]
```diff
@@ -89,8 +93,12 @@ Loader.prototype.start = function(onprogress, oncomplete) {
       var response = null;
       var request = new XMLHttpRequest();
       request.open('GET', Loader.rootPath + item.filepath, 1);
+      if(item.options.responseType) {
+        request.responseType = item.options.responseType;
+      }
       request.onload = function() {
-        item.callback(request.responseText);
+        item.callback(item.options.responseType == 'arraybuffer' ? request.response
```
Comment (iver56) [inline nin/dasBoot/Loader.js]: I get this error in Chrome/Opera, but not in FireFox:
Uncaught TypeError: item.callback is not a function

**Comment by iver56** [comment]
For reference, this is not compatible with IE11. Not that anyone cares ^^

**Comment by cristeahub** [comment]
> For reference, this is not compatible with IE11.

\o/

File: nin/frontend/app/scripts/controllers/bottom.js [hunk 2]
```diff
@@ -47,14 +47,16 @@
       $scope.loopEnd = null;
       $scope.loopActive = false;
       commands.on('setCuePoint', function() {
+        var currentBEAN = BEAN_FOR_FRAME($scope.currentFrame);
+        var currentQuantizedFrame = FRAME_FOR_BEAN(currentBEAN - currentBEAN % PROJECT.music.subdivision);
```
Comment (stianjensen) [inline nin/frontend/app/scripts/controllers/bottom.js]: Is this what I think it is? Loop points that snap to the music?

Comment (iver56) [inline nin/frontend/app/scripts/controllers/bottom.js]: Funny, that's actually a feature I thought of yesterday. Cool beans!

Comment (stianjensen) [inline nin/frontend/app/scripts/controllers/bottom.js]: Yeah, I tried creating it last week, but couldn't get the looping smooth enough with the HTML5 audio element.

Comment (stianjensen) [inline nin/frontend/app/scripts/controllers/bottom.js]: meowmeowbeenz

**Pull request merged**